spec/actors/hyku_addons/actors/json_fields_actor_spec.rb

```diff
@@ -0,0 +1,285 @@
+      end
+    end
+
+    context "with multiple valid json fields" do
+      let(:attributes) do
+        {
+          creator: [
+            { "creator_name_type" => "Personal", "creator_given_name" => "Liang", "creator_family_name" => "Gao" },
+            { "creator_name_type" => "Personal", "creator_given_name" => "Lidai", "creator_family_name" => "Wang" }
+          ],
+          editor: [
+            { "editor_name_type" => "Personal", "editor_given_name" => "Alexander A.", "editor_family_name" => "Oraevsky" },
+            { "editor_name_type" => "Personal", "editor_given_name" => "Lihong V.", "editor_family_name" => "Wang" }
+          ]
+        }
+      end
+
+      let(:transformed_attributes) do
+        { "creator" => ["[{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Liang\",\"creator_family_name\":\"Gao\"},{\"creator_name_type\":\"Personal\",\"creator_given_name\":\"Lidai\",\"creator_family_name\":\"Wang\"}]"], "editor" => ["[{\"editor_name_type\":\"Personal\",\"editor_given_name\":\"Alexander A.\",\"editor_family_name\":\"Oraevsky\"},{\"editor_name_type\":\"Personal\",\"editor_given_name\":\"Lihong V.\",\"editor_family_name\":\"Wang\"}]"], "contributor" => [], "funder" => [], "alternate_identifier" => [], "related_identifier" => [], "current_he_institution" => [] }
+      end
+
+      it "adds them all" do
+        expect(env.attributes).to eq(transformed_attributes)
+      end
+    end
+
+    context "with blank or empty attributes" do
+      let(:attributes) do
+        {
+          creator: [
+            {}
+          ],
+          editor: [
+            { "editor_name_type" => "Personal", "editor_given_name" => "Alexander A.", "editor_family_name" => nil }
+          ]
+        }
+      end
+
+      let(:transformed_attributes) do
+        { "editor" => ["[{\"editor_name_type\":\"Personal\",\"editor_given_name\":\"Alexander A.\",\"editor_family_name\":null}]"], "creator" => [], "contributor" => [], "funder" => [], "alternate_identifier" => [], "related_identifier" => [], "current_he_institution" => [] }
+      end
+
+      it "turns fields into null or adds empty arrays" do
+        expect(env.attributes).to eq(transformed_attributes)
+      end
+    end
+
+    context "with no name type" do
+      let(:attributes) do
+        {
+          editor: [
+            { "editor_name_type" => "", "editor_given_name" => "Lihong V.", "editor_family_name" => "Wang" }
+          ]
+        }
+      end
+
+      let(:transformed_attributes) do
+        { "editor" => ["[{\"editor_name_type\":\"\",\"editor_given_name\":\"Lihong V.\",\"editor_family_name\":\"Wang\"}]"], "creator" => [], "contributor" => [], "funder" => [], "alternate_identifier" => [], "related_identifier" => [], "current_he_institution" => [] }
+      end
+
+      it "adds the attributes" do
+        expect(env.attributes).to eq(transformed_attributes)
+      end
+    end
+
+    context "with nested fields" do
```

Comment:
I am not sure we have this use case of nesting creator but it is okay to have the test.

Reply:
I wanted to test the recursion. Can you think of an example where this might actually get used?

Reply:
If you mean an example where we might use nested creator, it is hard to think of one since the journals from where we get the creators don't have nested creators in as part of paper publications.

Also in over 2 years since this json feature was implemented and in use in the wild, that use case has not come up.